src/components/signup/SignupState.js

```diff
@@ -182,6 +193,7 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
     const paymentCode = _get(destinationPath, 'params.paymentCode')
 
     if (paymentCode || w3Token) {
+      fireEventByCode(APP_OPEN, { source: paymentCode ? 'payment code' : 'w3' })
```

Comment:
app_open shouldnt be fired here.
you move all app_open to init.js after the line where we initialize analytics
you can detect mobile
(window.matchMedia && window.matchMedia('(display-mode: standalone)').matches) ||
        window.navigator.standalone === true
payment code and w3 can be detected by parsing the window url

Reply:
done